src/compiler/compiler.ts

```diff
@@ -130,40 +132,44 @@ export class Compiler {
     }
 
     private runDiagnostics(program: ts.Program, host: ts.CompilerHost): void {
+        this.errors = [];
         let diagnostics = ts.getPreEmitDiagnostics(program);
         this.outputDiagnostics(diagnostics, host);
     }
 
     private outputDiagnostics(diagnostics: ts.Diagnostic[], host?: ts.FormatDiagnosticsHost): void {
 
-        if (diagnostics && diagnostics.length > 0) {
+        if (!diagnostics) {
```

Comment:
You probably know this, but just in case:
This `if` behaviour is different than before: now, if `diagnostics` is empty, it'll still run the rest of the function. Before, `diagnostics` needed to have at least one element to continue.

If this is on purpose, then no worries :)

Reply:
Nice catch 😄